development_server/src/main.rs

```diff
@@ -0,0 +1,313 @@
+        method,
+        headers,
+        body,
+        uri,
+    }
+}
+
+fn map_response(response: HttpResponse) -> Response<Body> {
+    trace!("mapping outgoing response {:#?}", response);
+    let mut builder = Response::builder().status(response.status);
+
+    let body = Body::from(response.body);
+
+    for (k, v) in response.headers.into_iter() {
+        builder = builder.header(k, v);
+    }
+
+    builder.body(body).expect("Failed to create a response")
+}
+
+// this is clearly abusing `map_request`
+// tower should have a way to "place a service"
+// in front of another one
+fn map_http_request(req: Request<Body>) -> HttpRequest {
+    tokio::task::block_in_place(|| {
+        let rt = tokio::runtime::Handle::current();
+        rt.block_on(map_request(req))
+    })
+}
+
+fn map_component_response(service: FunctionComponentService) -> HttpFunctionComponent {
+    let binding = service
+        .map_request(map_http_request)
+        .map_response(|resp| map_response(resp));
+    binding.boxed()
+}
+
+type HttpFunctionComponent = BoxService<Request<Body>, Response<Body>, BoxError>;
+
+type HttpFunctionComponentMaker = BoxCloneService<(), HttpFunctionComponent, BoxError>;
+
+fn new_http_component_maker(
+    component_maybe: Option<function_service::service::WasmComponent>,
+) -> HttpFunctionComponentMaker {
+    let store_producer = wasmtime_components::runtime::make_store_producer();
+    let base_component =
+        component_maybe.unwrap_or(function_service::service::empty_base_function_component());
+
+    new_function_service_maker(base_component, store_producer)
+        .map_response(|v| map_component_response(v))
+        .boxed_clone()
+}
+
+async fn start_development_server(
+    mut command_stream: ServiceCommandSource,
+    socket_addr: SocketAddr,
+) {
+    let (shutdown_tx, shutdown_rx) = tokio::sync::oneshot::channel();
+
+    let function_service_maker = Arc::new(Mutex::new(new_http_component_maker(None)));
+    let function_service_maker_rpc_handle = function_service_maker.clone();
+
+    let component_host_server = Server::bind(&socket_addr)
+        .serve(make_service_fn(move |v: &AddrStream| {
+            trace!("http connection {:#?}", v);
+            let cloned_maker = function_service_maker.clone();
+            async move {
+                let mut maker = cloned_maker.lock().await;
+                maker.call(()).await
+            }
+        }))
+        .with_graceful_shutdown(async move {
+            let _ = shutdown_rx.await;
+        });
+
+    // Spawn the manager event loop
+    let _ = tokio::spawn(async move {
+        let cloned_maker = function_service_maker_rpc_handle.clone();
+        while let Some(command) = command_stream.recv().await {
+            let _ = match command {
+                ServiceCommand::SwapFunctionComponent {
+                    component_path,
+                    reply,
+                } => {
+                    let component_path = Path::new(&component_path);
```

Comment:
Aha, perhaps I should use this approach for my PR in stead:
1. derive a Vec<String> based on contents read_dir
2. create paths based on those strings in stead of assigning them to a Guest struct